src/Nethermind/Nethermind.GitBook/RpcAndCliDataProvider.cs

```diff
@@ -0,0 +1,214 @@
+//  Copyright (c) 2021 Demerzel Solutions Limited
+//  This file is part of the Nethermind library.
+// 
+//  The Nethermind library is free software: you can redistribute it and/or modify
+//  it under the terms of the GNU Lesser General Public License as published by
+//  the Free Software Foundation, either version 3 of the License, or
+//  (at your option) any later version.a
+// 
+//  The Nethermind library is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+//  GNU Lesser General Public License for more details.
+// 
+//  You should have received a copy of the GNU Lesser General Public License
+//  along with the Nethermind. If not, see <http://www.gnu.org/licenses/>.
+// 
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using Nethermind.Cli.Modules;
+using Nethermind.JsonRpc.Modules;
+
+namespace Nethermind.GitBook
+{
+    public class RpcAndCliDataProvider
+    {
+        private Dictionary<string, Dictionary<string, MethodData>> _modulesData =
+            new Dictionary<string, Dictionary<string, MethodData>>();
+        
+
+        public Dictionary<string, Dictionary<string, MethodData>> GetRpcAndCliData()
+        {
+            List<Type> rpcTypes = GetRpcModules();
+            List<Type> cliTypes = GetCliModules();
+            AddRpcModulesData(rpcTypes);
+            AddCliModulesData(cliTypes);
+
+            return _modulesData;
+        }
+        
+        private List<Type> GetRpcModules()
+        {
```

Comment:
does it skip Baseline module and other plugins?

Reply:
Yes